dynawo/sources/Modeler/DataInterface/PowSyblIIDM/DYNDataInterfaceIIDM.h

```diff
@@ -21,44 +21,70 @@
 #define MODELER_DATAINTERFACE_POWSYBLIIDM_DYNDATAINTERFACEIIDM_H_
 
 #include "DYNDataInterface.h"
-#include "DYNSwitchInterface.h"
-#include "DYNShuntCompensatorInterface.h"
-#include "DYNDanglingLineInterface.h"
-#include "DYNStaticVarCompensatorInterface.h"
-#include "DYNTwoWTransformerInterface.h"
-#include "DYNThreeWTransformerInterface.h"
-#include "DYNVscConverterInterface.h"
-#include "DYNLccConverterInterface.h"
-#include "DYNHvdcLineInterface.h"
-#include "DYNLineInterface.h"
+#include "DYNSwitchInterfaceIIDM.h"
+#include "DYNShuntCompensatorInterfaceIIDM.h"
+#include "DYNDanglingLineInterfaceIIDM.h"
+#include "DYNStaticVarCompensatorInterfaceIIDM.h"
+#include "DYNTwoWTransformerInterfaceIIDM.h"
+#include "DYNThreeWTransformerInterfaceIIDM.h"
+#include "DYNVscConverterInterfaceIIDM.h"
+#include "DYNLccConverterInterfaceIIDM.h"
+#include "DYNHvdcLineInterfaceIIDM.h"
+#include "DYNLineInterfaceIIDM.h"
 #include "DYNCalculatedBusInterfaceIIDM.h"
+#include "DYNBusInterfaceIIDM.h"
+#include "DYNLoadInterfaceIIDM.h"
+#include "DYNGeneratorInterfaceIIDM.h"
 #include "DYNCriteria.h"
+#include "DYNNetworkInterfaceIIDM.h"
 #include "DYNServiceManagerInterfaceIIDM.h"
 #include <powsybl/iidm/Network.hpp>
 #include <boost/unordered_set.hpp>
 
+#include <mutex>
+
 namespace DYN {
 
 class DataInterfaceIIDM : public DataInterface {
  public:
   /**
    * @brief Build an instance of this class by reading a file
    * @param iidmFilePath iidm file path
+   * @param nbVariants number of variants of the network
    * @return The data interface built from the input file
    */
-  static boost::shared_ptr<DataInterface> build(std::string iidmFilePath);
+  static boost::shared_ptr<DataInterface> build(const std::string& iidmFilePath, unsigned int nbVariants = 1);
 
   /**
    * @brief Constructor
    * @param networkIIDM instance of iidm network
    */
-  explicit DataInterfaceIIDM(powsybl::iidm::Network&& networkIIDM);
+  explicit DataInterfaceIIDM(const boost::shared_ptr<powsybl::iidm::Network>& networkIIDM);
 
   /**
    * @brief Destructor
    */
   ~DataInterfaceIIDM();
 
+  /**
+   * @brief Copy constructor
+   *
+   * This does NOT clone criterias and IIDM network
```

Comment:
typo: criteria (might be at other places too)

Reply:
OK